samples/kvsWebRTCClientMaster.c

```diff
@@ -178,59 +243,81 @@ PVOID sendAudioPackets(PVOID args)
+    if(pSampleConfiguration == NULL) {
+        printf("[KVS Master] sendAudioPackets(): operation returned status code: 0x%08x \n", STATUS_NULL_ARG);
+        goto CleanUp;
+    }
 
     frame.presentationTs = 0;
 
     while (!ATOMIC_LOAD_BOOL(&pSampleConfiguration->appTerminateFlag)) {
         fileIndex = fileIndex % NUMBER_OF_OPUS_FRAME_FILES + 1;
-        SNPRINTF(filePath, MAX_PATH_LEN, "./opusSampleFrames/sample-%03d.opus", fileIndex);
-        CHK_STATUS(readFrameFromDisk(NULL, &frameSize, filePath));
+        snprintf(filePath, MAX_PATH_LEN, "./opusSampleFrames/sample-%03d.opus", fileIndex);
+
+        retStatus = readFrameFromDisk(NULL, &frameSize, filePath);
+        if(retStatus != STATUS_SUCCESS) {
+            printf("[KVS Master] readFrameFromDisk(): operation returned status code: 0x%08x \n", retStatus);
+            goto CleanUp;
+        }
 
         // Re-alloc if needed
         if (frameSize > pSampleConfiguration->audioBufferSize) {
-            CHK(NULL != (pSampleConfiguration->pAudioFrameBuffer = (PBYTE) MEMREALLOC(pSampleConfiguration->pAudioFrameBuffer, frameSize)), STATUS_NOT_ENOUGH_MEMORY);
-            pSampleConfiguration->audioBufferSize = frameSize;
+            pSampleConfiguration->pAudioFrameBuffer = (uint8_t*) realloc(pSampleConfiguration->pAudioFrameBuffer, frameSize);
+            if(pSampleConfiguration->pAudioFrameBuffer == NULL) {
+                printf("[KVS Master] Audio frame Buffer reallocation failed...%s (code %d)\n", strerror(errno), errno);
+                printf("[KVS Master] realloc(): operation returned status code: 0x%08x \n", STATUS_NOT_ENOUGH_MEMORY);
+                goto CleanUp;
+            }
         }
 
         frame.frameData = pSampleConfiguration->pAudioFrameBuffer;
         frame.size = frameSize;
 
-        CHK_STATUS(readFrameFromDisk(frame.frameData, &frameSize, filePath));
+        retStatus = readFrameFromDisk(frame.frameData, &frameSize, filePath);
+        if(retStatus != STATUS_SUCCESS) {
+            printf("[KVS Master] readFrameFromDisk(): operation returned status code: 0x%08x \n", retStatus);
+            goto CleanUp;
+        }
 
         frame.presentationTs += SAMPLE_AUDIO_FRAME_DURATION;
 
         if (!ATOMIC_LOAD_BOOL(&pSampleConfiguration->updatingSampleStreamingSessionList)) {
             ATOMIC_INCREMENT(&pSampleConfiguration->streamingSessionListReadingThreadCount);
             for (i = 0; i < pSampleConfiguration->streamingSessionCount; ++i) {
                 status = writeFrame(pSampleConfiguration->sampleStreamingSessionList[i]->pAudioRtcRtpTransceiver, &frame);
-                if (STATUS_FAILED(status)) {
-                    DLOGD("writeFrame failed with 0x%08x", status);
+                if (status != STATUS_SUCCESS) {
+                    printf("writeFrame failed with 0x%08x", status);
                 }
             }
             ATOMIC_DECREMENT(&pSampleConfiguration->streamingSessionListReadingThreadCount);
```

Comment:
Is this also a PIC macro?

Reply:
Yes. There are a few atomic operations throughout the file which are platform dependent and that is why I did not replace them yet. We need to think about how to deal with them.